scripts/synthetix/facilicate_trade.js

```diff
@@ -0,0 +1,84 @@
+    decmials: 18,
+  },
+  4: {
+    address: "0x0647b2C7a2a818276154b0fC79557F512B165bc1",
+    exchangeId: 12,
+    decmials: 18,
+  },
+}
+
+const sUSDByNetwork = {
+  1: {
+    address: "0x57Ab1E02fEE23774580C119740129eAC7081e9D3",
+    exchangeId: 9,
+    decmials: 18,
+  },
+  4: {
+    address: "0x1b642a124CDFa1E5835276A6ddAA6CFC4B35d52c",
+    exchangeId: 13,
+    decmials: 18,
+  },
+}
+
+module.exports = async callback => {
+  try {
+    const networkId = await web3.eth.net.getId()
+    const defaultAccount = (await web3.eth.getAccounts())[0]
+
+    const snxjs = new SynthetixJs({ networkId: networkId })
+    const exchange = await getExchange(web3)
+
+    const sETH = sETHByNetwork[networkId]
+    const sUSD = sUSDByNetwork[networkId]
+    // Both of these hardcoded tokens are assumed to have 18 decimal places.
+    // We "trust" that this will always be the case although it seems
+    // that synthetix reserves the authority to upgrade their token
+    // This could mean issuing a new one with a different number of decimals.
+    const sETHKey = ethers.utils.formatBytes32String("sETH")
+    const sUSDKey = ethers.utils.formatBytes32String("sUSD")
+
+    // Avoid querying exchange by tokenAddress for fixed tokenId
+    const buyTokens = [sETH, sUSD].map(token => token.exchangeId)
+    const sellTokens = [sUSD, sETH].map(token => token.exchangeId)
+
+    // Compute Rates and Fees based on price of sETH.
+    // Note that sUSD always has a price of 1 within synthetix protocol.
+    const exchangeRate = await snxjs.ExchangeRates.rateForCurrency(sETHKey)
+    const formatedRate = snxjs.utils.formatEther(exchangeRate)
+    console.log("sETH Price", snxjs.utils.formatEther(exchangeRate))
+
+    // Using synthetix's fees, and formatting their return values with their tools, plus parseFloat.
+    const sETHTosUSDFee = parseFloat(snxjs.utils.formatEther(await snxjs.Exchanger.feeRateForExchange(sETHKey, sUSDKey)))
+    const sUSDTosETHFee = parseFloat(snxjs.utils.formatEther(await snxjs.Exchanger.feeRateForExchange(sUSDKey, sETHKey)))
+
+    // Compute buy-sell amounts based on unlimited orders with rates from above.
+    const [lowerBuyAmount, lowerSellAmount] = calculateBuyAndSellAmountsFromPrice(formatedRate * (1 - sUSDTosETHFee), sUSD, sETH)
+    const [upperSellAmount, upperBuyAmount] = calculateBuyAndSellAmountsFromPrice(formatedRate * (1 + sETHTosUSDFee), sETH, sUSD)
+
+    const buyAmounts = [lowerBuyAmount, upperBuyAmount]
+    const sellAmounts = [lowerSellAmount, upperSellAmount]
+
+    // Fetch auction index and declare validity interval for orders.
+    // Note that order validity interval is inclusive on both sides.
+    const batchId = (await exchange.getCurrentBatchId.call()).toNumber()
+    const validFroms = Array(2).fill(batchId)
+    const validTos = Array(2).fill(batchId)
+
+    await exchange.placeValidFromOrders(buyTokens, sellTokens, validFroms, validTos, buyAmounts, sellAmounts, {
+      from: defaultAccount,
+    })
```

Comment:
The current batch id is used to create these orders. This means that these orders will appear at some point during the current batch. Another possibility is to create the order with batch id `batchId+1`, in this way the orders will be visible for the entire duration of the batch. Do we have some particular agreement with Synthetix about the timing of the price? Who provides the funds, and how?

Reply:
This won't be a concern for the use case here. Essentially people should assume that they can get the "going rate" in any batch so, in fact we might even place this order in the last minute of a batch so that nobody has a chance to have access to an old price estimate. 

In brief, it is intended that the orders are placed in the current batch.